### DISCUSSION TITLE
Unsafe.As, readonly version

### DISCUSSION THREAD

**Discussion by zugolbr** [dotnet/csharplang#4882] [Language Ideas] [answered]
Hi, we have
`ref TTo Unsafe.As<TFrom, TTo>(ref TFrom)`

Is there anything preventing
`ref readonly TTo Unsafe.ReadOnlyAs<TFrom, TTo>(ref readonly  TFrom)`

It would be quite handy... :)

**Comment by jnm2** [comment]
Sounds like a https://github.com/dotnet/runtime/discussions question. I know C# can't define overloads that differ only in `ref` vs `in` parameter modifiers, but these methods are [defined directly using CIL](https://github.com/dotnet/runtime/blob/v5.0.7/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il#L308-L314) rather than C#, so the real question would be whether C# is able to resolve an overload at consumption time.

**Reply by huoyaoyuan** [reply]
Yes exactly. https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE+UkALIgLIAUATgKYBmiAlgHYAu9AlNrljLogL4c4BeZETKN6zNkK49+MXkA

**Reply by zugolbr** [reply]
Well, it would be nice too if it was called Unsafe.ReadOnlyAs(...), there is no real need that they share the same name...
(just updated my original question)
I'll post the same question in https://github.com/dotnet/runtime/discussions, thanks.

**Comment by zugolbr** [comment, marked as answer]
Question transferred to https://github.com/dotnet/runtime/discussions/54789.
